localization/pose_estimator_arbiter/src/pose_estimator_arbiter/pose_estimator_arbiter_core.cpp

```diff
@@ -0,0 +1,202 @@
+    declare_parameter<std::vector<std::string>>("pose_sources"), get_logger())),
+  logger_configure_(std::make_unique<tier4_autoware_utils::LoggerLevelConfigure>(this))
+{
+  // Shared data
+  shared_data_ = std::make_shared<SharedData>();
+
+  // Publisher
+  pub_diag_ = create_publisher<DiagnosticArray>("/diagnostics", 10);
+  pub_debug_string_ = create_publisher<String>("~/debug/string", 10);
+  pub_debug_marker_array_ = create_publisher<MarkerArray>("~/debug/marker_array", 10);
+
+  // Stoppers
+  for (auto pose_estimator_name : running_estimator_list_) {
+    switch (pose_estimator_name) {
+      case PoseEstimatorType::ndt:
+        stoppers_.emplace(
+          pose_estimator_name, std::make_shared<stopper::StopperNdt>(this, shared_data_));
+        break;
+      case PoseEstimatorType::yabloc:
+        stoppers_.emplace(
+          pose_estimator_name, std::make_shared<stopper::StopperYabLoc>(this, shared_data_));
+        break;
+      case PoseEstimatorType::eagleye:
+        stoppers_.emplace(
+          pose_estimator_name, std::make_shared<stopper::StopperEagleye>(this, shared_data_));
+        break;
+      case PoseEstimatorType::artag:
+        stoppers_.emplace(
+          pose_estimator_name, std::make_shared<stopper::StopperArTag>(this, shared_data_));
+        break;
+      default:
+        RCLCPP_ERROR_STREAM(get_logger(), "invalid pose_estimator is specified");
+    }
+  }
+  {
+    using std::placeholders::_1;
+    const rclcpp::QoS sensor_qos = rclcpp::SensorDataQoS();
+    const rclcpp::QoS latch_qos = rclcpp::QoS(1).transient_local().reliable();
+
+    // Subscriber for stoppers
+    sub_artag_input_ = create_subscription<Image>(
+      "~/input/artag/image", sensor_qos, shared_data_->artag_input_image.create_callback());
+    sub_yabloc_input_ = create_subscription<Image>(
+      "~/input/yabloc/image", sensor_qos, shared_data_->yabloc_input_image.create_callback());
+    sub_ndt_input_ = create_subscription<PointCloud2>(
+      "~/input/ndt/pointcloud", sensor_qos, shared_data_->ndt_input_points.create_callback());
+    sub_eagleye_output_ = create_subscription<PoseCovStamped>(
+      "~/input/eagleye/pose_with_covariance", 5,
+      shared_data_->eagleye_output_pose_cov.create_callback());
+
+    // Subscriber for switch rule
+    sub_localization_pose_cov_ = create_subscription<PoseCovStamped>(
+      "~/input/pose_with_covariance", 5, shared_data_->localization_pose_cov.create_callback());
+    sub_point_cloud_map_ = create_subscription<PointCloud2>(
+      "~/input/pointcloud_map", latch_qos, shared_data_->point_cloud_map.create_callback());
+    sub_vector_map_ = create_subscription<HADMapBin>(
+      "~/input/vector_map", latch_qos, shared_data_->vector_map.create_callback());
+    sub_initialization_state_ = create_subscription<InitializationState>(
+      "~/input/initialization_state", latch_qos,
+      shared_data_->initialization_state.create_callback());
+  }
```

Comment:
Is it necessary to subscribe all of these even when Autoware doesn't use all pose_estimators?

For example, `/localization/eagleye/pose_with_covariance/to_relay` appears in `ros2 topic list` even when `pose_source:=ndt_yabloc`, and pose_estimator_arbiter will subscribe `/sensing/camera/traffic_light/image_raw` even when `pose_source:=ndt_eagleye`.

Is it possible to declare them only when the according pose_estimator_name exists?

Reply:
As you suggest, it is not necessary to always subscribe to all topics.  :+1: 
I modified it in https://github.com/autowarefoundation/autoware.universe/pull/5846/commits/f358d0d00b8dc0b78093895e6fa45df5945da7c8

With this commit, a subscriber is created only when the corresponding pose_estimator is running. 
(It also contains a minor bug fix. never mind)

Reply:
Thank you for your modification!
I confirmed that no extra topics were subscribed when they are not called for.